Simplify logic to resolve tasks stuck in queued despite stalled_task_timeout (#30375)
mistercrunch/Airflow: newsfragments/30375.significant.rst

> Simplify logic to resolve tasks stuck in queued despite stalled_task_timeout

Reviewer: ```suggestion
Consolidate handling of tasks stuck in queued under new ``task_queued_timeout`` config
```

This is probably a better title?

Logic for handling tasks stuck in the queued state has been consolidated, and the all configurations 
responsible for timing out stuck queued tasks have been deprecated and merged into 

> `scheduler.task_queued_timeout`. The configurations that have been deprecated are 

Reviewer: ```suggestion
`[scheduler] task_queued_timeout`. The configurations that have been deprecated are 
```

Let's reference these config options like this instead.

> Logic for handling tasks stuck in the queued state has been consolidated, and the all configurations 
> responsible for timing out stuck queued tasks have been deprecated and merged into 
> `scheduler.task_queued_timeout`. The configurations that have been deprecated are 
> `kubernetes.worker_pods_pending_timeout`, `celery.stalled_task_timeout`, and 
> `celery.task_adoption_timeout`. If any of these configurations are set, the longest timeout will be
> respected. For example, if `celery.stalled_task_timeout` is 1200, and `scheduler.task_queued_timeout` 
> is 600, Airflow will set `scheduler.task_queued_timeout` to 1200.

Reviewer: ```suggestion
Logic for handling tasks stuck in the queued state has been consolidated, and the all configurations
responsible for timing out stuck queued tasks have been deprecated and merged into
``scheduler.task_queued_timeout``. The configurations that have been deprecated are
``kubernetes.worker_pods_pending_timeout``, ``celery.stalled_task_timeout``, and
``celery.task_adoption_timeout``. If any of these configurations are set, the longest timeout will be
respected. For example, if ``celery.stalled_task_timeout`` is 1200, and ``scheduler.task_queued_timeout``
is 600, Airflow will set ``scheduler.task_queued_timeout`` to 1200.

```

Fixes static checks.
